### PR TITLE
Add isEmpty utility method

### PR DESCRIPTION
**Pull request by TJKoury**
Added isEmpty utility method. Needed a more efficient way to check if a BitField instance has no set bits, using early return for positives.

### CODE REVIEW

File: src/index.ts [hunk 1]
```diff
@@ -27,15 +27,23 @@ export default class BitField {
...
-        return this.buffer.length >> 3;
+        return this.buffer.length << 3;
     }
 
+    /** Variable to store the count of bits set to 1 */
+    private setBitCount: number = 0;
+
+    /** Set to keep track of indices of set bits */
+    private readonly setBits: Set<number> = new Set();
+
```
Comment (fb55) [inline src/index.ts:44]: I would like to keep the memory usage of bitfield the same as it is, and especially the set will require quite a bit more memory. They are also not needed for `isEmpty`. Let's remove them?

File: package.json [hunk 2]
```diff
@@ -1,7 +1,7 @@
 {
     "name": "bitfield",
     "description": "a simple bitfield, compliant with the BitTorrent spec",
-    "version": "4.1.0",
+    "version": "4.2.0",
```
Comment (fb55) [inline package.json:4]: `npm version` bumps this, please don't change it manually.

**Review by fb55 — COMMENTED** [review]
I'd love to add `isEmpty`. The PR also adds a bunch of other methods that will make memory usage increase a lot and that I don't want to add to this module.

**Comment by fb55** [comment]
I went ahead and addressed my review comments. Thanks @TJKoury!

**Comment by TJKoury** [comment]
Awesome!  Thanks for accepting; those other methods help with time complexity but at the cost of memory (as always is the trade-off).

**Pull request merged**